workflows/prognostic_c48_run/runtime/factories.py

```diff
@@ -24,3 +27,37 @@ def get_emulator_adapter(
             diagnostic_variables=set(config.diagnostic_variables),
             timestep=timestep,
         )
+
+
+def get_overrider_adapter(
+    config: UserConfig,
+    state: DerivedFV3State,
+    communicator: fv3gfs.util.CubedSphereCommunicator,
+    timestep: float,
+) -> Optional[OverriderAdapter]:
+    if config.overrider is None:
+        return None
+    else:
+        return OverriderAdapter(
+            config.overrider,
+            state,
+            communicator,
+            timestep,
+            diagnostic_variables=set(config.diagnostic_variables),
+        )
+
+
+def get_fv3_physics_adapter(
+    config: UserConfig,
+    state: DerivedFV3State,
+    timestep: float,
+    communicator: fv3gfs.util.CubedSphereCommunicator,
+) -> Optional[Union[PrognosticAdapter, OverriderAdapter]]:
+    if config.online_emulator is not None and config.overrider is not None:
```

Comment:
It seems it should be straightforward to both prescribe and emulate at the same time. I could envision wanting to do this for a some kind of piggy-backing setup.

Otherwise, I would use an union type for the config so that it is actually impossible to configure the model incorrectly.

Reply:
Yeah I suppose it's true we might want to do both at once. I can adjust it so they are composable.

Reply:
Now if a user provides emulation and tendency prescribing configs, both will be done. (emulation first, which is then overriden with the prescribed tendency. Seems more likely we'd want to do it in that order than the reverse).